addons/movments_accounts_balances/controllers/controllers.py

```diff
@@ -199,6 +199,198 @@ def delete_bill(self, bill_id):
         except Exception as e:
             return "Failed to delete bill: {}".format(e)
 
+    ##create/get/delete_bills payment
+    @api.model
+    def create_bill_payment(self, bill_ids, journal_id, payment_date, payment_method_line_id):
+        AccountMove = self.env['account.move']
+        bills = AccountMove.browse(bill_ids)
+
+        if not bills.exists():
+            raise UserError("No valid bills found with the provided IDs.")
+        if any(bill.state != 'posted' for bill in bills):
+            raise UserError("All bills must be posted to proceed with payment.")
+
+        payment_register = self.env['account.payment.register'].with_context({
+            'active_model': 'account.move',
+            'active_ids': bill_ids,
+        }).create({
+            'payment_date': payment_date,
+            'journal_id': journal_id,
+            'payment_method_line_id': payment_method_line_id,
+        })
+
+        payment_register.action_create_payments()
+
+        payment_data_list = []  # Initialize an empty list to store payment data
+
+        for bill in bills:
+            # Adjust the lambda to the correct field and value for 'payable' account types
+            account_move_lines_to_reconcile = bill.line_ids.filtered(
+                lambda line: not line.reconciled and line.account_id.account_type == 'payable'
+            )
+
+            # Search for move lines linked to this bill that are not yet reconciled
+            move_lines_to_reconcile = self.env['account.move.line'].search([
```

Comment:
@sameerz09 are these `move_lines_to_reconcile` yet to be reconciled or are they actually already reconciled?

Since you are searching for `reconciled=True` I think they are already reconciled?

Reply:
If they are already reconciled, please rename to "reconciled_move_lines"